Altis_Life.Altis/core/fn_initMedic.sqf

```diff
@@ -14,13 +14,18 @@ if ((FETCH_CONST(life_medicLevel)) < 1 && (FETCH_CONST(life_adminlevel) isEqualT
     sleep 35;
 };
 
+// -- Restrict Weapons
 if (LIFE_SETTINGS(getNumber,"restrict_medic_weapons") isEqualTo 1) then {
-    [] spawn {
-        for "_i" from 0 to 1 step 0 do {
-            waitUntil {sleep 3; !(currentWeapon player isEqualTo "")};
-            removeAllWeapons player;
+    private _firedEVH = player addEventHandler ["Fired, {
+        // -- Get Params
+        params ["_unit", "", "", "_weapon"];
+        
+        // -- Check if weapon is equipped
+        if (!(_weapon _unit isEqualTo "")) then {
+            // -- Remove Weapon from Player
+            removeWeapon _weapon player;
```

Comment:
I would guess, if you are restricting them picking the gun up, they won't be able to fire.